components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineSession.kt

```diff
@@ -312,7 +312,7 @@ class SystemEngineSession(private val defaultSettings: Settings? = null) : Engin
 
         return webView?.let {
             it.buildDrawingCache()
-            val outBitmap = Bitmap.createBitmap(webView.drawingCache)
+            val outBitmap = webView.drawingCache?.let { Bitmap.createBitmap(it) }
```

Comment:
Is it the same if we just use **it**?
```kotlin
            val outBitmap = Bitmap.createBitmap(it.drawingCache)
```

Reply:
@Amejia481 this is a different *it*. Every let block has its own implicit *it*. The one you are referring to is the outer let block where *it* is the Webview, this is the inner let block where *it* is the non-null drawing cache.

Reply:
I can write this differently to make it easier to read.

Reply:
I renamed it in the inner block, so that should be easier to read.

Reply:
Exactly.
What I meant was that we can rewrite the code using **it**.
```kotlin
 override fun captureThumbnail(): Bitmap? {
        val webView = currentView()
        return webView?.let {
            it.buildDrawingCache()
            val outBitmap = Bitmap.createBitmap(it.drawingCache)
            it.destroyDrawingCache()
            outBitmap
        }
    }
```

Reply:
@Amejia481 no, we can't, because it wasn't the WebView that was null, but the drawingCache. So, this would still cause the same problem when the drawingCache is null.

Reply:
But, we can/should do this:
`it.drawingCache?.let { cache -> Bitmap.createBitmap(cache) }`

Reply:
Got it!